File: service/backend/src/main/java/ru/tykvin/hermes/security/CurrentUserHolder.java
package ru.tykvin.hermes.security;

import org.springframework.stereotype.Component;
import ru.tykvin.hermes.model.User;

@Component
public class CurrentUserHolder {
    private static final ThreadLocal<User> USER_THREAD_LOCAL = new ThreadLocal<>();
    public void set(User user) {
        USER_THREAD_LOCAL.set(user);
    }

    public User get() {
        return USER_THREAD_LOCAL.get();
    }
}

File: src/main/kotlin/ru/tykvin/hermes/auth/service/CurrentUserHolder.kt
package ru.tykvin.hermes.auth.service

import org.springframework.stereotype.Component
import ru.tykvin.hermes.auth.controller.User

@Component
class CurrentUserHolder {

    fun set(user: User) {
        USER_THREAD_LOCAL.set(user)
    }

    fun get(): User {
        return USER_THREAD_LOCAL.get()
    }

    companion object {
        private val USER_THREAD_LOCAL = ThreadLocal<User>()
    }
}
